datafusion/common/src/dfschema.rs

```diff
@@ -281,6 +282,18 @@ impl DFSchema {
             .all(|(dffield, arrowfield)| dffield.name() == arrowfield.name())
     }
 
+    /// Check to see if fields in 2 Arrow schemas are compatible
+    pub fn check_arrow_schema_type_compatible(&self, arrow_schema: &Schema) -> bool {
+        let self_arrow_schema: Schema = self.into();
+        self_arrow_schema
+            .fields()
+            .iter()
+            .zip(arrow_schema.fields().iter())
+            .all(|(l_field, r_field)| {
+                can_cast_types(r_field.data_type(), l_field.data_type())
```

Comment:
If this would return a Result<()> we could return a more specific error of the type mismatch here (and pass it to the user).

Reply:
@Dandandan OK, I'll fix it.